contracts/actions/curveusd/CurveUsdPayback.sol

```diff
@@ -0,0 +1,124 @@
+        params.from = _parseParamAddr(params.from, _paramMapping[1], _subData, _returnValues);
+        params.onBehalfOf = _parseParamAddr(params.onBehalfOf, _paramMapping[2], _subData, _returnValues);
+        params.to = _parseParamAddr(params.to, _paramMapping[3], _subData, _returnValues);
+        params.paybackAmount = _parseParamUint(params.paybackAmount, _paramMapping[4], _subData, _returnValues);
+        params.maxActiveBand = int256(_parseParamUint(uint256(params.maxActiveBand), _paramMapping[5], _subData, _returnValues));
+
+        (uint256 paybackAmount, bytes memory logData) = _curveUsdPayback(params);
+        emit ActionEvent("CurveUsdPayback", logData);
+        return bytes32(paybackAmount);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes memory _callData) public payable virtual override {
+        Params memory params = parseInputs(_callData);
+
+        (, bytes memory logData) = _curveUsdPayback(params);
+        logger.logActionDirectEvent("CurveUsdPayback", logData);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    function _curveUsdPayback(Params memory _params) internal returns (uint256, bytes memory) {
+        /// @dev see ICrvUsdController natspec
+        if (_params.paybackAmount == 0) revert ZeroAmountPayback();
+        
+        if (!isControllerValid(_params.controllerAddress)) revert CurveUsdInvalidController();
+
+        if (_params.onBehalfOf == address(0)) _params.onBehalfOf = address(this);
+
+        /// @dev debtAmount > debt is safe, need to make sure we dont pull more than needed
+        /// @dev this also closes the position
+        bool isClose;
+        uint256 debt = ICrvUsdController(_params.controllerAddress).debt(_params.onBehalfOf);
+        
+        if (_params.paybackAmount >= debt) {
+            _params.paybackAmount = debt;
+            isClose = true;
+        }
+
+        _params.paybackAmount = CRVUSD_TOKEN_ADDR.pullTokensIfNeeded(_params.from, _params.paybackAmount);
+        CRVUSD_TOKEN_ADDR.approveToken(_params.controllerAddress, _params.paybackAmount);
+
+        address collateralAsset = ICrvUsdController(_params.controllerAddress).collateral_token();
+
+
+        uint256 startingBaseCollBalance;
+        uint256 startingCrvUsdBalance;
+        if (isClose) {
+            startingBaseCollBalance = collateralAsset.getBalance(address(this));
+            startingCrvUsdBalance = CRVUSD_TOKEN_ADDR.getBalance(address(this)) - debt;
+        }
+
+        if (collateralAsset == TokenUtils.WETH_ADDR){
+            ICrvUsdController(_params.controllerAddress).repay(_params.paybackAmount, _params.onBehalfOf, _params.maxActiveBand, false);
```

Comment:
Can't we use this version where the last param is `false` in all the tokens, no need for the weth special handle?